keras_cv/layers/preprocessing/random_zoom.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+from keras import backend
+
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+from keras_cv.utils import preprocessing
+
+# In order to support both unbatched and batched inputs, the horizontal
+# and verticle axis is reverse indexed
+H_AXIS = -3
+W_AXIS = -2
+IMAGES = "images"
+BOUNDING_BOXES = "bounding_boxes"
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomZoom(BaseImageAugmentationLayer):
+    """A preprocessing layer which randomly zooms images during training.
+
+    This layer will randomly zoom in or out on each axis of an image
+    independently, filling empty space according to `fill_mode`.
+
+    Input pixel values can be of any range (e.g. `[0., 1.)` or `[0, 255]`) and
+    of interger or floating point dtype. By default, the layer will output
+    floats.
+
+    Args:
+      height_factor: a float represented as fraction of value, or a tuple of
+        size 2 representing lower and upper bound for zooming vertically. When
+        represented as a single float, this value is used for both the upper and
+        lower bound. A positive value means zooming out, while a negative value
+        means zooming in. For instance, `height_factor=(0.2, 0.3)` result in an
+        output zoomed out by a random amount in the range `[+20%, +30%]`.
+        `height_factor=(-0.3, -0.2)` result in an output zoomed in by a random
+        amount in the range `[+20%, +30%]`.
```

Comment:
Seems like this range is wrong -- should it be [-30%, -20%]?

Reply:
done